Tests/SnapshotTests/Components/TwintComponentUITests.swift

```diff
@@ -0,0 +1,210 @@
+
+        let didSubmitExpectation = expectation(description: "PaymentComponentDelegate must be called.")
+        paymentComponentDelegate.onDidSubmit = { data, component in
+            XCTAssertTrue(component === self.sut)
+
+            // Then
+            XCTAssertTrue(data.storePaymentMethod ?? false)
+
+            didSubmitExpectation.fulfill()
+        }
+
+        // When
+        let view = sut.viewController.view
+        let submitButton: UIControl = try XCTUnwrap(view?.findView(by: TwintViewIdentifier.submitButton))
+        submitButton.sendActions(for: .touchUpInside)
+
+        wait(for: [didSubmitExpectation], timeout: 10)
+    }
+
+    func testSubmitStoreDetailsIsEnabledShouldSendPaymentComponentDataWithStorePaymentMethodTrue() throws {
+        // Given
+        let configuration = TwintComponentConfiguration(showsStorePaymentMethodField: true, storePaymentMethod: false)
+        sut = TwintComponent(
+            paymentMethod: paymentMethod,
+            context: context,
+            configuration: configuration
+        )
+
+        let paymentComponentDelegate = PaymentComponentDelegateMock()
+        sut.delegate = paymentComponentDelegate
+
+        setupRootViewController(sut.viewController)
+
+        let didSubmitExpectation = expectation(description: "PaymentComponentDelegate must be called.")
+        paymentComponentDelegate.onDidSubmit = { data, component in
+            XCTAssertTrue(component === self.sut)
+
+            // Then
+            XCTAssertTrue(data.storePaymentMethod ?? false)
+
+            didSubmitExpectation.fulfill()
+        }
+
+        // When
+        let view = sut.viewController.view
+        let storeDetailsToggle: FormToggleItemView = try XCTUnwrap(view?.findView(by: TwintViewIdentifier.storeDetailsToogle))
+        let submitButton: UIControl = try XCTUnwrap(view?.findView(by: TwintViewIdentifier.submitButton))
+
+        storeDetailsToggle.item.value = true
+        submitButton.sendActions(for: .touchUpInside)
+
+        wait(for: [didSubmitExpectation], timeout: 10)
+    }
+
+    func testSubmitStoreDetailsIsDisabledShouldSendPaymentComponentDataWithStorePaymentMethodFalse() throws {
+        // Given
+        let configuration = TwintComponentConfiguration(showsStorePaymentMethodField: true, storePaymentMethod: false)
+        sut = TwintComponent(
+            paymentMethod: paymentMethod,
+            context: context,
+            configuration: configuration
+        )
+
+        let paymentComponentDelegate = PaymentComponentDelegateMock()
+        sut.delegate = paymentComponentDelegate
+
+        setupRootViewController(sut.viewController)
+
+        let didSubmitExpectation = expectation(description: "PaymentComponentDelegate must be called.")
+        paymentComponentDelegate.onDidSubmit = { data, component in
+            XCTAssertTrue(component === self.sut)
+
+            // Then
+            XCTAssertFalse(data.storePaymentMethod ?? false)
```

Comment:
This should be `?? true` but I'd suggest using force wrap or XCTUnwrap instead of `??`